client_connection/manager.go

```diff
@@ -38,15 +39,27 @@ func NewManager(mysteriumClient server.Client, dialogEstablisherFactory DialogEs
 	}
 }
 
-func (manager *connectionManager) Connect(identity identity.Identity, NodeKey string) error {
+func (manager *connectionManager) Connect(identity identity.Identity, nodeKey string) error {
 	manager.status = statusConnecting()
-	clientSession, err := manager.mysteriumClient.SessionCreate(NodeKey)
+
+	proposals, err := manager.mysteriumClient.Proposals(nodeKey)
 	if err != nil {
 		manager.status = statusError(err)
 		return err
 	}
+	if len(proposals) == 0 {
+		err = errors.New("node has no service proposals")
+		manager.status = statusError(err)
+		return err
+	}
+	proposal := proposals[0]
 
-	proposal := clientSession.ServiceProposal
+	// TODO: pass in selected proposal to session creation
+	clientSession, err := manager.mysteriumClient.SessionCreate(nodeKey)
```

Comment:
This is not needed and should go away. Client gets session id from node with last dialog message (VpnSession at the moment)